### ISSUE TITLE
[2.1.x]Can't build with php5.6

### ISSUE THREAD

**Issue by ucando** [phalcon/cphalcon#11961]
when I run build/install, it cause fatal error: ext/standard/php_smart_string.h: No such file or directory.

**Comment by Jurigag** [comment]
It's really weird. Can you try to build it using zephir ? Just install zephir and then:

```
zephir fullclean
zephir build
```

Also you sure you have `php5-dev` ? I just don't know why it looks for php_smart_string.h in php 5.6 build. It's file from php 7, in php 5 it's called php_smart_str.h ? @andresgutierrez

**Comment by ucando** [comment]
I have another old source, last commit id is dacf42c0d45ebb3b9bf6832d14c74cf207426698, this can build success.
I think there are some codes about php7 support has problem.

**Comment by ucando** [comment]
I try to compile with zephir, it works.

**Comment by CrNix** [comment]
I have the same issue. I am also able to build using Zephir. But after downloading latest repo, compiling the C source is not working with an error referencing php_smart_string.h file.

(using Linux Mint distro)

How do you run the Phalcon tests? I navigated into build/php5/64bit and run make test but it ended up with error as well.

Thank you

**Comment by Jurigag** [comment]
Did you typed zephir fullclean before ?

**Comment by CrNix** [comment]
Fullclean? Here is my full list of commands:
````
cphalcon $ zephir fullclean
cphalcon $ zephir build
````
Here comes bunch of compile messages, warnings that  some mcrypt functions are not available at compile time...
After the message "Restarting Apache" comes the Segmentation fault error.
But not a big deal, Phalcon module is installed correctly and works OK in CLI mode.
````
cphalcon $ zephir fullclean
cphalcon $ cd ./build/php5/64bit
cphalcon/build/php5/64bit $ make test
````
And this fails...
If you can give a hint what to do to run the tests, I would be very gratefull.

**Comment by Jurigag** [comment]
There are commands for running tests:

https://github.com/phalcon/cphalcon/blob/2.1.x/.travis.yml#L79

**Comment by CrNix** [comment]
Thank you for your help. Unfortunately I do not have much experience with Travis CI. But now it seems the tests are working.

**Comment by Jurigag** [comment]
You don't need to use travis ci. Just run those commands are in this file.

**Comment by Jurigag** [comment]
@andresgutierrez 

Line 18765 in `build/php5/64bits/phalcon.zep.c`

```
#include <ext/standard/php_smart_string.h>
#include <zend_smart_str.h>
```

Shouldn't be there:

```
#if PHP_VERSION_ID < 70000
#include <ext/standard/php_smart_str.h>
#else
#include <ext/standard/php_smart_string.h>
#include <zend_smart_str.h>
#endif
```

Like in other places ?

**Comment by virgofx** [comment]
@andresgutierrez Can we get a PR to address the installation issues

1) Fix the `./build/install` to work for PHP5.4, 5.5, 5.6
2) Add consistency to Phalcon installation for all versions - Presently, there is via Zephir for 2.1x/PHP7 and ./build/install. There is also the /ext folder as well which add confusion for a lot of people.  The website currently references `./build/install`. Can we just clean this all up and make it one way for everything, reference that everywhere, and then ensure that travis builds all 4 versions (5.4, 5.5, 5.6, 7.0) as there are a lot of production systems that build from branch that have been affected by this as well.

**Comment by Jurigag** [comment]
@virgofx travis is just using zephir, that's why everything works on CI but people have problems beacause are using `./build/install`.

**Comment by virgofx** [comment]
See point #2 above -- we need to ensure that there is consistency 100% everywhere so builds, CI, production, documentation are all consistent - whether that's Travis or manual bash scripts , doesn't matter .. Just need to get consistent.

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.
